config/opal_check_libnl.m4

```diff
@@ -0,0 +1,262 @@
+                          [opal_libnlv3_happy=0])])
+    LIBS=$LIBS_save
+    LDFLAGS=$LDFLAGS_save
+
+    # Just because libnl* is evil, double check that the
+    # netlink/version.h we found was for libnl v3.  As far as we
+    # know, netlink/version.h only first appeared in version
+    # 3... but let's really be sure.
+    AS_IF([test $opal_libnlv3_happy -eq 1],
+          [AC_MSG_CHECKING([to ensure these really are libnl v3 headers])
+           AS_IF([test -n "$$2_CPPFLAGS"],
+                 [CPPFLAGS="$$2_CPPFLAGS $CPPFLAGS"])
+           AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+#include <netlink/netlink.h>
+#include <netlink/version.h>
+#ifndef LIBNL_VER_MAJ
+#error "LIBNL_VER_MAJ not defined!"
+#endif
+/* to the best of our knowledge, version.h only exists in libnl v3 */
+#if LIBNL_VER_MAJ != 3
+#error "LIBNL_VER_MAJ != 3, I am sad"
+#endif
+                              ]])],
+                             [AC_MSG_RESULT([yes])],
+                             [AC_MSG_RESULT([no])
+                              opal_libnlv3_happy=0])])
+
+    CPPFLAGS=$CPPFLAGS_save
+
+    # If we found everything
+    AS_IF([test $opal_libnlv3_happy -eq 1],
+          [$2_LIBS="-lnl-3 -lnl-route-3"
+           OPAL_HAVE_LIBNL3=1])
+
+   OPAL_VAR_SCOPE_POP
+])
+
+dnl
+dnl Check for libnl.
+dnl
+dnl Inputs:
+dnl
+dnl $1: prefix where to look for libnl
+dnl $2: var name prefix of _CPPFLAGS and _LDFLAGS and _LIBS
+dnl
+dnl Outputs:
+dnl
+dnl - Set $2_CPPFLAGS necessary to compile with libnl
+dnl - Set $2_LDFLAGS necessary to link with libnl
+dnl - Set $2_LIBS necessary to link with libnl
+dnl - Set OPAL_HAVE_LIBNL3 0 if libnl will be used
+dnl
+AC_DEFUN([OPAL_CHECK_LIBNL_V1],[
+    AC_MSG_NOTICE([checking for libnl v1])
+
+    OPAL_CHECK_PACKAGE([$2],
+                       [netlink/netlink.h],
+                       [nl],
+                       [nl_connect],
+                       [-lm],
+                       [$1],
+                       [],
+                       [opal_libnlv1_happy=1],
+                       [opal_libnlv1_happy=0])
+
+    AS_IF([test $opal_libnlv1_happy -eq 1],
+          [$2_LIBS="-lnl -lm"
+           OPAL_HAVE_LIBNL3=0])
+])
+
+dnl
+dnl Summarize libnl and libnl3 usage,
+dnl and abort if conflict is found
+dnl
+dnl Print the list of libraries that use libnl, 
+dnl the list of libraries that use libnl3,
+dnl and aborts if both libnl and libnl3 are used.
+dnl
+AC_DEFUN([OPAL_CHECK_LIBNL_SUMMARY],[
+    AC_MSG_CHECKING([for libraries that use libnl v1])
+    AS_IF([test -n "$opal_libnlv1_libs"],
+          [AC_MSG_RESULT([$opal_libnlv1_libs])],
+          [AC_MSG_RESULT([(none)])])
+    AC_MSG_CHECKING([for libraries that use libnl v3])
+    AS_IF([test -n "$opal_libnlv3_libs"],
+          [AC_MSG_RESULT([$opal_libnlv3_libs])],
+          [AC_MSG_RESULT([(none)])])
+    AS_IF([test -n "$opal_libnlv1_libs" && test -n "$opal_libnlv3_libs"],
+          [AC_MSG_WARN([Both libnl v1 and libnl v3 are required])
```

Comment:
The use of the word "required" here is a bit confusing -- a user could read that as "to build Open MPI, it's telling me that both libnl v1 and v3 are required, but then it also tells me that this is an error!"

Maybe s/are required/have been found as dependent libraries/ ...?